### PR TITLE
fix: set logs to False if wait is False

### PR DESCRIPTION
**Pull request by ChuyangDeng**
*Issue #, if available:*
#1584 

*Description of changes:*
set `logs` to False when `wait` is False, instead of raising an error

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to any/all clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/automl/automl.py [hunk 1]
```diff
@@ -78,16 +78,13 @@ def fit(self, inputs=None, wait=True, logs=True, job_name=None):
                 is stored. Or an AutoMLInput object. If a local path is provided, the dataset will
                 be uploaded to an S3 location.
             wait (bool): Whether the call should wait until the job completes (default: True).
-            logs (bool): Whether to show the logs produced by the job.
-                Only meaningful when wait is True (default: True).
+            logs (bool): Whether to show the logs produced by the job. Only meaningful when wait
+                is True (default: True). if `wait` is False, `log` will be set to False as well.
```
Comment (veerabio) [inline src/sagemaker/automl/automl.py:82]: typo: `logs` missing **s**
 if `wait` is False, `log**s**` will be set to False as well.

**Comment by metrizable** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-notebook-tests
* Commit ID: c066263c2dd0d080f001afe3d29cfdc9e84ab6b1
* Result: FAILED
* [Build Logs](https://3rpiknuts4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=96c6aeb8-275c-44a7-bd85-dcc51b4e5e43%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/sagemaker/automl/automl.py (hunk 1 above)
Comment (ChuyangDeng) [reply src/sagemaker/automl/automl.py]: Thanks! updated!

**Comment by ChuyangDeng** [comment]
> unit test?

I updated the test (also add a message to warn that logs is going to be set to False when wait is False but logs is True)

File: src/sagemaker/automl/automl.py [hunk 2]
```diff
@@ -78,16 +81,14 @@ def fit(self, inputs=None, wait=True, logs=True, job_name=None):
                 is stored. Or an AutoMLInput object. If a local path is provided, the dataset will
                 be uploaded to an S3 location.
             wait (bool): Whether the call should wait until the job completes (default: True).
-            logs (bool): Whether to show the logs produced by the job.
-                Only meaningful when wait is True (default: True).
+            logs (bool): Whether to show the logs produced by the job. Only meaningful when wait
+                is True (default: True). if `wait` is False, `logs` will be set to False as well.
```
Comment (laurenyu) [inline src/sagemaker/automl/automl.py:85]: nit: need two backticks for formatting

File: src/sagemaker/automl/automl.py [hunk 3]
```diff
@@ -78,16 +81,14 @@ def fit(self, inputs=None, wait=True, logs=True, job_name=None):
...
                 a default job name, based on the training image name and current timestamp.
         """
-        if logs and not wait:
-            raise ValueError(
-                """Logs can only be shown if wait is set to True.
-                Please either set wait to True or set logs to False."""
-            )
+        if not wait and logs:
+            logs = False
+            logger.warning("logs will be set to False. logs is only meaningful when wait is True.")
```
Comment (laurenyu) [inline src/sagemaker/automl/automl.py:91]: s/logs will be set/Setting logs

File: tests/unit/sagemaker/automl/test_auto_ml.py [hunk 4]
```diff
@@ -348,6 +348,7 @@ def test_auto_ml_additional_optional_params(sagemaker_session):
         "generate_candidate_definitions_only": GENERATE_CANDIDATE_DEFINITIONS_ONLY,
         "tags": TAGS,
     }
+    assert "logs will be set to False. logs is only meaningful when wait is True." in caplog.text
```
Comment (laurenyu) [inline tests/unit/sagemaker/automl/test_auto_ml.py:351]: in addition to testing for the log output, you could `patch("sagemaker.automl.AutoMLJob.start_new")` and then call `.wait.assert_not_called()` on the resulting mock

Comment (ChuyangDeng) [reply tests/unit/sagemaker/automl/test_auto_ml.py]: added a new test instead to test warning messages and `assert_not_called()`

**Pull request merged**